### PR TITLE
Fix clang-tidy 14 mingw issues (fixes #5612)

### LINKED ISSUE #5612: Clang-tidy 14 issues
Some more issues that I found with clang-tidy 14 but didn't fix in #5611 because I'm not entirely sure about them.

Clang-Tidy LLVM 14.0.3 with MinGW64 (gcc 12.1.0) on Windows 10 at a971a16ed527984dbe147ee882a6f6974669ca2f.

<details>
<summary>
clang-analyzer-core.StackAddressEscape
</summary>

```
ddnet/src/engine/server.h:132:4: warning: Address of stack memory associated with local variable 'aBuf' is still referred to by the stack variable 'tmp' upon returning to the caller.  This will be a dangling reference [clang-analyzer-core.StackAddressEscape]
                        return SendPackMsgOne(&Msg7, Flags, ClientID);
                        ^
ddnet/src/game/server/gamecontext.cpp:4084:5: note: Assuming the condition is false
        if(pFilter[0])
           ^~~~~~~~~~
ddnet/src/game/server/gamecontext.cpp:4084:2: note: Taking false branch
        if(pFilter[0])
        ^
ddnet/src/game/server/gamecontext.cpp:4088:2: note: Calling 'CGameContext::SendChatTarget'
        SendChatTarget(ClientID, aBuf);
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
ddnet/src/game/server/gamecontext.cpp:401:5: note: Assuming field 'm_SvDemoChat' is not equal to 0
        if(g_Config.m_SvDemoChat)
           ^~~~~~~~~~~~~~~~~~~~~
ddnet/src/game/server/gamecontext.cpp:401:2: note: Taking true branch
        if(g_Config.m_SvDemoChat)
        ^
ddnet/src/game/server/gamecontext.cpp:402:3: note: Calling 'IServer::SendPackMsg'
                Server()->SendPackMsg(&Msg, MSGFLAG_VITAL | MSGFLAG_NOSEND, -1);
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
ddnet/src/engine/server.h:71:3: note: Taking true branch
                if(ClientID == -1)
                ^
ddnet/src/engine/server.h:73:19: note: Assuming the condition is true
                        for(int i = 0; i < MaxClients(); i++)
                                       ^~~~~~~~~~~~~~~~
ddnet/src/engine/server.h:73:4: note: Loop condition is true.  Entering loop body
                        for(int i = 0; i < MaxClients(); i++)
                        ^
ddnet/src/engine/server.h:74:8: note: Assuming the condition is true
                                if(ClientIngame(i))
                                   ^~~~~~~~~~~~~~~
ddnet/src/engine/server.h:74:5: note: Taking true branch
                                if(ClientIngame(i))
                                ^
ddnet/src/engine/server.h:77:15: note: Calling 'IServer::SendPackMsgTranslate'
                                        Result = SendPackMsgTranslate(&tmp, Flags, i);
                                                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
ddnet/src/engine/server.h:118:6: note: Assuming field 'm_ClientID' is >= 0
                if(pMsg->m_ClientID >= 0 && !Translate(pMsg->m_ClientID, ClientID))
                   ^~~~~~~~~~~~~~~~~~~~~
ddnet/src/engine/server.h:118:6: note: Left side of '&&' is true
ddnet/src/engine/server.h:118:31: note: Assuming the condition is true
                if(pMsg->m_ClientID >= 0 && !Translate(pMsg->m_ClientID, ClientID))
                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
ddnet/src/engine/server.h:118:3: note: Taking true branch
                if(pMsg->m_ClientID >= 0 && !Translate(pMsg->m_ClientID, ClientID))
                ^
ddnet/src/engine/server.h:125:6: note: Assuming the condition is true
                if(IsSixup(ClientID))
                   ^~~~~~~~~~~~~~~~~
ddnet/src/engine/server.h:125:3: note: Taking true branch
                if(IsSixup(ClientID))
                ^
ddnet/src/engine/server.h:130:18: note: Assuming field 'm_Team' is <= 0
                        Msg7.m_Mode = pMsg->m_Team > 0 ? protocol7::CHAT_TEAM : protocol7::CHAT_ALL;
                                      ^~~~~~~~~~~~~~~~
ddnet/src/engine/server.h:130:18: note: '?' condition is false
ddnet/src/engine/server.h:132:4: note: Address of stack memory associated with local variable 'aBuf' is still referred to by the stack variable 'tmp' upon returning to the caller.  This will be a dangling reference
                        return SendPackMsgOne(&Msg7, Flags, ClientID);
                        ^
ddnet/src/engine/server.h:135:3: warning: Address of stack memory associated with local variable 'aBuf' is still referred to by the stack variable 'tmp' upon returning to the caller.  This will be a dangling reference [clang-analyzer-core.StackAddressEscape]
                return SendPackMsgOne(pMsg, Flags, ClientID);
                ^
ddnet/src/game/server/gamecontext.cpp:4084:5: note: Assuming the condition is false
        if(pFilter[0])
           ^~~~~~~~~~
ddnet/src/game/server/gamecontext.cpp:4084:2: note: Taking false branch
        if(pFilter[0])
        ^
ddnet/src/game/server/gamecontext.cpp:4088:2: note: Calling 'CGameContext::SendChatTarget'
        SendChatTarget(ClientID, aBuf);
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
ddnet/src/game/server/gamecontext.cpp:401:5: note: Assuming field 'm_SvDemoChat' is not equal to 0
        if(g_Config.m_SvDemoChat)
           ^~~~~~~~~~~~~~~~~~~~~
ddnet/src/game/server/gamecontext.cpp:401:2: note: Taking true branch
        if(g_Config.m_SvDemoChat)
        ^
ddnet/src/game/server/gamecontext.cpp:402:3: note: Calling 'IServer::SendPackMsg'
                Server()->SendPackMsg(&Msg, MSGFLAG_VITAL | MSGFLAG_NOSEND, -1);
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
ddnet/src/engine/server.h:71:3: note: Taking true branch
                if(ClientID == -1)
                ^
ddnet/src/engine/server.h:73:19: note: Assuming the condition is true
                        for(int i = 0; i < MaxClients(); i++)
                                       ^~~~~~~~~~~~~~~~
ddnet/src/engine/server.h:73:4: note: Loop condition is true.  Entering loop body
                        for(int i = 0; i < MaxClients(); i++)
                        ^
ddnet/src/engine/server.h:74:8: note: Assuming the condition is true
                                if(ClientIngame(i))
                                   ^~~~~~~~~~~~~~~
ddnet/src/engine/server.h:74:5: note: Taking true branch
                                if(ClientIngame(i))
                                ^
ddnet/src/engine/server.h:77:15: note: Calling 'IServer::SendPackMsgTranslate'
                                        Result = SendPackMsgTranslate(&tmp, Flags, i);
                                                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
ddnet/src/engine/server.h:118:6: note: Assuming field 'm_ClientID' is >= 0
                if(pMsg->m_ClientID >= 0 && !Translate(pMsg->m_ClientID, ClientID))
                   ^~~~~~~~~~~~~~~~~~~~~
ddnet/src/engine/server.h:118:6: note: Left side of '&&' is true
ddnet/src/engine/server.h:118:31: note: Assuming the condition is true
                if(pMsg->m_ClientID >= 0 && !Translate(pMsg->m_ClientID, ClientID))
                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
ddnet/src/engine/server.h:118:3: note: Taking true branch
                if(pMsg->m_ClientID >= 0 && !Translate(pMsg->m_ClientID, ClientID))
                ^
ddnet/src/engine/server.h:125:6: note: Assuming the condition is false
                if(IsSixup(ClientID))
                   ^~~~~~~~~~~~~~~~~
ddnet/src/engine/server.h:125:3: note: Taking false branch
                if(IsSixup(ClientID))
                ^
ddnet/src/engine/server.h:135:3: note: Address of stack memory associated with local variable 'aBuf' is still referred to by the stack variable 'tmp' upon returning to the caller.  This will be a dangling reference
                return SendPackMsgOne(pMsg, Flags, ClientID);
                ^
```

</details>

<details>
<summary>
cppcoreguidelines-virtual-class-destructor
</summary>

```
ddnet/src/test/aio.cpp:8:7: warning: destructor of 'Async' is protected and virtual [cppcoreguidelines-virtual-class-destructor]
class Async : public ::testing::Test
      ^
ddnet/src/test/aio.cpp:8:7: note: make it protected and non-virtual
class Async : public ::testing::Test
      ^
```

</details>

<details>
<summary>
clang-analyzer-deadcode.DeadStores (false positive, but I don't know why clang-tidy doesn't detect the usage in dbg_assert)
</summary>

```
ddnet/src/engine/client/text.cpp:1968:5: warning: Value stored to 'HasNonEmptyTextContainer' is never read [clang-analyzer-deadcode.DeadStores]
                                HasNonEmptyTextContainer = true;
                                ^                          ~~~~
ddnet/src/engine/client/text.cpp:1968:5: note: Value stored to 'HasNonEmptyTextContainer' is never read
                                HasNonEmptyTextContainer = true;
                                ^                          ~~~~
```

</details>

<details>
<summary>
modernize-deprecated-headers
</summary>

```
ddnet/src/base/system.cpp:65:10: warning: inclusion of deprecated C++ header 'errno.h'; consider using 'cerrno' instead [modernize-deprecated-headers]
#include <errno.h>
         ^~~~~~~~~
         <cerrno>
```

</details>

<details>
<summary>
clang-analyzer-optin.portability.UnixAPI (probably a false positive because it doesn't know that argc will always be at least 1)
</summary>

```
ddnet/src/base/system.cpp:3753:24: warning: Call to 'malloc' has an allocation size of 0 bytes [clang-analyzer-optin.portability.UnixAPI]
        new_argv[0] = (char *)malloc(total_size);
                              ^      ~~~~~~~~~~
ddnet/src/base/system.cpp:3743:2: note: 'total_size' initialized to 0
        int total_size = 0;
        ^~~~~~~~~~~~~~
ddnet/src/base/system.cpp:3745:17: note: Assuming 'i' is >= 'wide_argc'
        for(int i = 0; i < wide_argc; i++)
                       ^~~~~~~~~~~~~
ddnet/src/base/system.cpp:3745:2: note: Loop condition is false. Execution continues on line 3752
        for(int i = 0; i < wide_argc; i++)
        ^
ddnet/src/base/system.cpp:3753:24: note: Call to 'malloc' has an allocation size of 0 bytes
        new_argv[0] = (char *)malloc(total_size);
                              ^      ~~~~~~~~~~
```

</details>

<details>
<summary>
clang-diagnostic-error (everywhere where SDL is included, looks like a problem with SDL code)
</summary>

```
ddnet/ddnet-libs/sdl/include/windows/SDL_stdinc.h:708:12: error: cannot initialize return object of type 'void *' with an rvalue of type 'void' [clang-diagnostic-error]
    return SDL_memcpy(dst, src, dwords * 4);
           ^
ddnet/ddnet-libs/sdl/include/windows/SDL_stdinc.h:678:20: note: expanded from macro 'SDL_memcpy'
#define SDL_memcpy memcpy
                   ^
ddnet/ddnet-libs/sdl/include/windows/SDL_stdinc.h:708:12: error: use of undeclared identifier 'memcpy'; did you mean 'mem_copy'? [clang-diagnostic-error]
    return SDL_memcpy(dst, src, dwords * 4);
           ^
ddnet/ddnet-libs/sdl/include/windows/SDL_stdinc.h:678:20: note: expanded from macro 'SDL_memcpy'
#define SDL_memcpy memcpy
                   ^
ddnet/src/base/system.h:124:6: note: 'mem_copy' declared here
void mem_copy(void *dest, const void *source, unsigned size);
     ^
```

</details>

### PR DESCRIPTION
**Pull request by def-**
Strangely I didn't get all of the errors.

## Checklist

- [ ] Tested the change ingame
- [ ] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test (especially base/) or added coverage to integration test
- [ ] Considered possible null pointers and out of bounds array indexing
- [ ] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

**Comment by def-** [comment]
Hm, it seems to work, but then we run into other problems when linking, which means we basically link against two crypto libs at once. Do we want that?
```
/usr/bin/ld: warning: libcrypto.so.1.1, needed by /usr/lib/libcurl.so, may conflict with libcrypto.so.3
```

**Comment by heinrich5991** [comment]
> I think I prefered ignoring the warning for now.

Ignoring the warning does seem preferable.



> Hm, it seems to work, but then we run into other problems when linking, which means we basically link against two crypto libs at once. Do we want that?

Where do you see that ld warning?

**Comment by def-** [comment]
On every executable we link.

File: src/engine/server.h [hunk 1]
```diff
@@ -129,10 +129,10 @@ class IServer : public IInterface
 			Msg7.m_pMessage = pMsg->m_pMessage;
 			Msg7.m_Mode = pMsg->m_Team > 0 ? protocol7::CHAT_TEAM : protocol7::CHAT_ALL;
 			Msg7.m_TargetID = -1;
-			return SendPackMsgOne(&Msg7, Flags, ClientID);
+			return SendPackMsgOne(&Msg7, Flags, ClientID); // NOLINT(clang-analyzer-core.StackAddressEscape)
```
Comment (Chairn) [inline src/engine/server.h:132]: I don't think we should ignore this one. Looks like a potential bug when stack is claimed by another function or variable.

Comment (Robyt3) [reply src/engine/server.h:132]: See #5624.

Comment (def-) [reply src/engine/server.h]: Removed as I approved #5624

**Review by Chairn — APPROVED** [review]

**Comment by heinrich5991** [comment]
Accidental update of `ddnet-libs`?

**Comment by def-** [comment]
Rebased, which also fixed the ddnet-libs

**Comment by heinrich5991** [comment]
bors r+

Thanks.

**Pull request merged**